backend/src/webhook/elopage.ts

```diff
@@ -0,0 +1,370 @@
+        loginElopgaeBuy.productPrice = parseFloat(val)
+        break
+      case 'payer[email]':
+        loginElopgaeBuy.payerEmail = val
+        break
+      case 'publisher[email]':
+        loginElopgaeBuy.publisherEmail = val
+        break
+      case 'payment_state':
+        loginElopgaeBuy.payed = val === 'paid'
+        break
+      case 'success_date':
+        loginElopgaeBuy.successDate = new Date(val)
+        break
+      case 'event':
+        loginElopgaeBuy.event = val
+        break
+      case 'membership[id]':
+        // TODO this was never set on login_server - its unclear if this is the correct value
+        loginElopgaeBuy.elopageUserId = parseInt(val)
+        break
+      case 'payer[first_name]':
+        firstName = val
+        break
+      case 'payer[last_name]':
+        lastName = val
+        break
+      default:
+        // eslint-disable-next-line no-console
+        console.log(`Unknown Elopage Value '${entry}'`)
+    }
+    return null // we write things into the loginElopgaeBuy object, no return value needed
+  })
+
+  // Do not process certain events
+  if (['lesson.viewed', 'lesson.completed', 'lesson.commented'].includes(loginElopgaeBuy.event)) {
+    // eslint-disable-next-line no-console
+    console.log('User viewed, completed or commented - not saving hook')
+    return
+  }
+
+  // Save the hook data
+  await loginElopgaeBuy.save()
+
+  // create user for certain products
+  /*
+    Registrierung - Schritt 1 von 3,  36001
+    Gradido-Basis,                    43741
+    Premium-Mitgliedschaft,           43870
+    Gold-Mitgliedschaft,              43944
+    Business-Mitgliedschaft,          43960
+    Förderbeitrag:                    49106
+  */
+  if ([36001, 43741, 43870, 43944, 43960, 49106].includes(loginElopgaeBuy.productId)) {
+    const email = loginElopgaeBuy.payerEmail
+
+    const VALIDATE_EMAIL = /^[a-zA-Z0-9.!#$%&?*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/
+    const VALIDATE_NAME = /^<>&;]{2,}$/
+
+    // Validate inputs
+    if (
+      email === '' ||
+      !email.match(VALIDATE_EMAIL) ||
+      firstName === '' ||
+      firstName.match(VALIDATE_NAME) ||
+      lastName === '' ||
+      lastName.match(VALIDATE_NAME)
+    ) {
+      // eslint-disable-next-line no-console
+      console.log(`Could not create User ${firstName} ${lastName} with email: ${email}`)
+      return
+    }
+
+    // Do we already have such a user?
+    if ((await LoginUser.count({ email })) !== 0) {
+      // eslint-disable-next-line no-console
+      console.log(`Did not create User - already exists with email: ${email}`)
+      return
+    }
+
+    // generate a random password - 8 random bytes, the email, special char, capital & small letter, number and another set of 8 random bytes
+    // TODO: The user will be forced to reset his password - how was this done before?
+    const password =
+      randomBytes(8).toString('hex') + email + '!aA1' + randomBytes(8).toString('hex')
```

Comment:
Good solution to ensure a valid password!

Reply:
Not good enough tho. We need to have a special case for the users coming from elopage where they have to set their password after email activation

Reply:
We should think about changing the whole process. See #1123